examples/Vesting.purs

```diff
@@ -0,0 +1,147 @@
+-- | This module demonstrates how the `Contract` interface can be used to build,
+-- | balance, and submit transactions. It creates two transactions: one that 
+-- | locks some Ada in a script address and one unlocks and claims the same Ada 
+-- | back to the owner's address by submitting a transaction that is signed by
+-- | the correct PublicKeyHash after the locking period has expired.
+
+module Ctl.Examples.Vesting (main, example, contract) where
+
+import Contract.Prelude
+
+import Contract.Address
+  ( PaymentPubKeyHash
+  , Slot
+  , getNetworkId
+  , nativeScriptHashEnterpriseAddress
+  , ownPaymentPubKeyHash
+  , ownStakePubKeyHash
+  )
+import Contract.Chain (currentSlot, waitNSlots)
+import Contract.Config (ConfigParams, testnetNamiConfig)
+import Contract.Hashing (nativeScriptHash)
+import Contract.Log (logInfo')
+import Contract.Monad
+  ( Contract
+  , launchAff_
+  , liftContractE
+  , liftContractM
+  , liftedM
+  , runContract
+  )
+import Contract.Numeric.Natural (fromInt')
+import Contract.ScriptLookups as Lookups
+import Contract.Test.E2E (publishTestFeedback)
+import Contract.Time
+  ( from
+  , futureSlot
+  , getEraSummaries
+  , getSystemStart
+  , slotToPosixTime
+  )
+import Contract.Transaction
+  ( NativeScript(ScriptAll, ScriptPubkey, TimelockStart)
+  , TransactionHash
+  , TransactionInput
+  , _input
+  , awaitTxConfirmed
+  , lookupTxHash
+  )
+import Contract.TxConstraints as Constraints
+import Contract.Utxos (utxosAt)
+import Contract.Value as Value
+import Ctl.Examples.Helpers
+  ( buildBalanceSignAndSubmitTx
+  , mustPayToPubKeyStakeAddress
+  ) as Helpers
+import Data.Array ((!!))
+import Data.BigInt as BigInt
+import Data.Lens (view)
+import Data.Map as Map
+import Data.Newtype (unwrap)
+
+main :: Effect Unit
+main = example testnetNamiConfig
+
+nativeScript ∷ PaymentPubKeyHash → Slot → NativeScript
+nativeScript pkh slot = ScriptAll
+  [ ScriptPubkey $ unwrap $ unwrap pkh
+  , TimelockStart slot
```

Comment:
shouldn't this be `TimelockExpiry slot`? but then again the example works, so maybe I'm getting something wrong..